ibis/backends/dask/execution/maps.py

```diff
@@ -0,0 +1,178 @@
+import collections
+import functools
+
+import dask.dataframe as dd
+import numpy as np
+import pandas
+
+import ibis.expr.operations as ops
+from ibis.backends.pandas.execution.maps import (
+    execute_node,
+    safe_get,
+    safe_keys,
+    safe_merge,
+    safe_values,
+)
+
+
+@execute_node.register(ops.MapLength, dd.Series)
+def execute_map_length_series(op, data, **kwargs):
+    return data.map(len, na_action='ignore')
+
+
+@execute_node.register(ops.MapValueForKey, dd.Series, dd.Series)
+def execute_map_value_for_key_series_series(op, data, key, **kwargs):
+    assert data.size == key.size, 'data.size != key.size'
+    return data.map(
+        lambda x, keyiter=iter(key.values): x.get(next(keyiter), None)
+    )
+
+
+def none_filled_dask_series(n):
+    dd.from_array(np.full(n, None))
+
+
+@execute_node.register(ops.MapValueForKey, dd.Series, type(None))
+def execute_map_value_for_key_series_none(op, data, key, **kwargs):
+    return none_filled_dask_series(len(data))
+
+
+@execute_node.register(ops.MapValueForKey, dd.Series, object)
+def execute_map_value_for_key_series_scalar(op, data, key, **kwargs):
+    return data.map(functools.partial(safe_get, key=key))
+
+
+@execute_node.register(ops.MapValueForKey, collections.abc.Mapping, dd.Series)
+def execute_map_value_for_key_dict_series(op, data, key, **kwargs):
+    return key.map(functools.partial(safe_get, data))
+
+
+@execute_node.register(ops.MapValueOrDefaultForKey, dd.Series, object, object)
+def map_value_default_series_scalar_scalar(op, data, key, default, **kwargs):
+    return data.map(functools.partial(safe_get, key=key, default=default))
+
+
+@execute_node.register(
+    ops.MapValueOrDefaultForKey, dd.Series, object, dd.Series
+)
+def map_value_default_series_scalar_series(op, data, key, default, **kwargs):
+    return data.map(
+        lambda mapping, key=key, defaultiter=iter(default.values): safe_get(
+            mapping, key, next(defaultiter)
+        )
+    )
+
+
+@execute_node.register(
+    ops.MapValueOrDefaultForKey, dd.Series, dd.Series, object
+)
+def map_value_default_series_series_scalar(op, data, key, default, **kwargs):
+    return data.map(
+        lambda mapping, keyiter=iter(key.values), default=default: safe_get(
+            mapping, next(keyiter), default
+        )
+    )
+
+
+@execute_node.register(
+    ops.MapValueOrDefaultForKey, dd.Series, dd.Series, dd.Series
+)
+def execute_map_value_default_series_series_series(op, data, key, default):
+    def get(
+        mapping, keyiter=iter(key.values), defaultiter=iter(default.values)
+    ):
+        return safe_get(mapping, next(keyiter), next(defaultiter))
+
+    return data.map(get)
+
+
+# Note - to avoid dispatch ambiguities we must unregister pandas
```

Comment:
really? why? these are a dd.Series should override here? (now a way around this is to in a common location you simply register with both types, e.g. in a dask/pandas ops section). we can do that as a followup, though maybe makes sense to split it out now (before this PR).

Reply:
If you don't unregister these, https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/tests/test_core.py#L44 will fail. If I understand the issue correct it is that `multipledisptach` sees 

```
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, pandas.Series, object,)
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, object, pandas.Series,)
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, dd.Series, object,)
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, object, dd.Series,)
```
as an ambiguous set of registrations. via https://multiple-dispatch.readthedocs.io/en/latest/resolution.html#ambiguities.

Reply:
hmm, this looks like we should actually just register both (e.g. do this as a common set). 

though why are these ambiguous?

Reply:
Good question - that's the source of the failing pandas test as well. I _think_ what is happening is `multipledispatch` sees two ambiguities:

```python
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, object, pandas.Series,)
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, dd.Series, object,)
```
and separately 
```python
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, pandas.Series, object,)
(ops.MapValueOrDefaultForKey, collections.abc.Mapping, object, dd.Series,)
```
`multipledispatch` sees this as ambiguous because it won't know what do do if you call this with a `(pandas.Series, dd.Series)` object. I've changed the registration around in https://github.com/ibis-project/ibis/pull/2557/commits/c125fe753137c9b5adffcf54833446466c2d029a to both avoid the ambiguity and keep dispatch working correctly in pandas (even if you have also loaded the dask backend at the same time). I tried to clarify in the code why we were doing it this way. 

Let me know if this was a reasonable approach and the commentary is clear, or you want this done a different way.